enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,474 @@
+    ingressNetworkLoadBalancer:
+      eipAllocations:
+      - eipalloc-1111
+      - eipalloc-2222
+      - eipalloc-3333
+      - eipalloc-4444
+      - eipalloc-5555 
+...
+```  
+
+2. The installer will create the cluster ingress config object as follows:
+```yaml
+ oc edit ingress.config/cluster -o yaml
+ ...
+ apiVersion: config.openshift.io/v1
+ kind: Ingress
+ metadata:
+...
+   name: cluster
+...
+ spec:
+   domain: apps.eip.devcluster.openshift.com
+   loadBalancer:
+     platform:
+       aws:
+         type: NLB
+         eipAllocations:
+           ingressNetworkLoadBalancer:
+           - eipalloc-1234567890abcdef1
+           - eipalloc-1234567890abcdef2
+           - eipalloc-1234567890abcdef3
+           - eipalloc-1234567890abcdef4
+           - eipalloc-1234567890abcdef5
+       type: AWS
+```
+3. Ingress Operator will create the `default` `IngressController` with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` with values from the field `eipAllocations`
+   from the cluster ingress config object.
+
+#### Updating the default IngressController's EIPs after installation:
+
+The cluster-admin will need to update the eipAllocations in `IngressConfig` and then delete the `default IngressController`. 
+The cluster-ingress-operator will create a new `default IngressController` with the newly updated eipAllocations.
+
+### API Extensions
+
+This proposal doesn't add any API extensions other than the new field proposed in
+[IngressConfig API Updates](#ingressconfig-api-updates).
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+1. #### Set EIP through installer for the default IngressController:
+    1. The admin sets the `eipAllocations` in the installer using the install-config.yaml.
+    2. The installer then sets the ingress cluster object with the `eipAllocations`.
+    3. The cluster ingress operator then creates a `default` `IngressController` by:
+        1. creating an `default` `IngressController` CR.
+        2. then scaling a deployment for the `IngressController` CR which has the value of the `eipAllocations` from the
+           ingress cluster object.
+        3. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`,
+           which uses the value from the field `eipAllocations` of `IngressController` CR.
+
+2. #### Validation on installer when installing in managed VPC (full-automated) based in the discovered zones used to create the cluster.
```

Comment:
All the Validation subsections belong in one section entitled "Field Validation".

I'm not exactly sure what you mean here, but maybe this:
```suggestion
#### Field Validation
##### The number of Availability Zones much match the number of `eipAllocations`

The installer must ensure this when installing in managed VPC (full-automated).  It can compare the number of `eipAllocations` to the discovered zones.
```

Reply:
Added Field Validation